### PR TITLE
RavenDB-25485 Fix AI Agent Handle not working when returning a list

### PR DESCRIPTION
**Pull request by ayende**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-25485

### Additional description

`AddActionResponse<TResponse>` used `ToBlittable()` which internally uses `BlittableJsonWriter` — it only supports root-level JSON objects. When a Handle callback returns a collection (`List<T>`, arrays), Newtonsoft serializes it as a JSON array, causing the writer's state machine to throw `InvalidOperationException`.

Fix: use the Newtonsoft serializer directly to a cached `StringWriter`/`StringBuilder`, which handles any type correctly. Applied to both `AddActionResponse` and `AddArtificialActionWithResponse`.

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [x] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing
- [ ] Existing tests verify the correct behavior

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/AI/AiConversation.cs [hunk 1]
```diff
@@ -332,6 +325,16 @@ private async Task<AiAnswer<TAnswer>> RunAsyncInternal<TAnswer>(string streamPro
         }
     }
 
+    private string SerializeToJson(object value)
+    {
+        _jsonBuffer ??= new StringBuilder();
+        _jsonWriter ??= new StringWriter(_jsonBuffer);
+        _jsonBuffer.Clear();
+        var serializer = (JsonSerializer)_aiOperations._store.Conventions.Serialization.CreateSerializer();
```
[the remark this answers is not in the document]
Comment (ayende) [reply src/Raven.Client/Documents/AI/AiConversation.cs:333]: We don't have another such implementation right now, so not an issue

File: src/Raven.Client/Documents/AI/AiConversation.cs [hunk 2]
```diff
@@ -332,6 +325,16 @@ private async Task<AiAnswer<TAnswer>> RunAsyncInternal<TAnswer>(string streamPro
...
+        serializer.Serialize(_jsonWriter, value);
+        return _jsonBuffer.ToString();
```
[the remark this answers is not in the document]
Comment (ayende) [reply src/Raven.Client/Documents/AI/AiConversation.cs:335]: yes, this is by design, conversations aren't typically long lived. Just to reduce allocations if you have a lot of actions calls.

**Comment by ayende** [comment]
> * Declaring a new private field named _settings hides the protected _settings field from AbstractOpenAiChatCompletionClientSettings. Even with the explicit `new`, member hiding here is easy to trip over during future edits; consider renaming this field (e.g. _azureSettings) or exposing a typed property to avoid ambiguity.

This is intentional, since we expose the same _instance_, once as the base class settings in the base class, and again as teh concrete settings type in the derived class

File: src/Raven.Server/Documents/AI/Settings/OpenAiChatCompletionClientSettings.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ namespace Raven.Server.Documents.AI.Settings;
 
 internal class OpenAiChatCompletionClientSettings : AbstractOpenAiChatCompletionClientSettings
 {
-    private readonly OpenAiSettings _settings;
+    private new readonly OpenAiSettings _settings;
```
[the remark this answers is not in the document]
Comment (ayende) [reply src/Raven.Server/Documents/AI/Settings/OpenAiChatCompletionClientSettings.cs:13]: This is intentional, since we expose the same instance, once as the base class settings in the base class, and again as teh concrete settings type in the derived class

File: src/Raven.Server/Documents/AI/Settings/AzureOpenAiChatCompletionClientSettings.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ namespace Raven.Server.Documents.AI.Settings;
 
 internal class AzureOpenAiChatCompletionClientSettings : AbstractOpenAiChatCompletionClientSettings
 {
-    private readonly AzureOpenAiSettings _settings;
+    private new readonly AzureOpenAiSettings _settings;
```
[the remark this answers is not in the document]
Comment (ayende) [reply src/Raven.Server/Documents/AI/Settings/AzureOpenAiChatCompletionClientSettings.cs:12]: This is intentional, since we expose the same instance, once as the base class settings in the base class, and again as teh concrete settings type in the derived class

File: test/FastTests/Issues/RavenDB-25485.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    [RavenFact(RavenTestCategory.Ai)]
+    public void AddActionResponseShouldWorkWhenReturningAnEmptyList()
+    {
+        using var store = GetDocumentStore();
+
+        var chat = store.AI.Conversation("agents/test", "chats/123", new AiConversationCreationOptions());
+
+        chat.AddActionResponse("tool-1", new List<ProductInfo>());
```
Comment (golan22888) [inline test/FastTests/Issues/RavenDB-25485.cs:59]: Do we also want to assert the output itself here? So the test verifies correctness of the produced JSON and not only that the call succeeds(especially for the empty list case)?

Comment (arekpalinski) [reply test/FastTests/Issues/RavenDB-25485.cs:59]: @ayende please check the comment

Comment (ayende) [reply test/FastTests/Issues/RavenDB-25485.cs:76]: Good call, fixed

**Review by golan22888 — APPROVED** [review]
